great_tables/_utils_render_html.py

```diff
@@ -318,7 +329,7 @@ def create_columns_component_h(data: GTData) -> str:
                         colspan=1,
                         style=_flatten_styles(styles_column_labels + styles_i),
                         scope="col",
-                        id=_process_text_id(remaining_headings_labels[j]),
+                        id=_create_element_id(table_id, remaining_headings_labels[j]),
```

Comment:
To complete the move to using column IDs instead of labels for the element ID, I think you need to do this:

```suggestion
                        id=_create_element_id(table_id, remaining_headings_ids[j]),
```

To make this work, you'll need to add:

```python
remaining_heading_ids = [entry.var for entry in boxhead if entry.var in remaining_headings]
```

after the definition of remaining_headings_labels on line 305.